eva/utils/math_utils.py

```diff
@@ -0,0 +1,48 @@
+# coding=utf-8
+# Copyright 2018-2022 EVA
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import numpy as np
+
+
+def get_centroid(bboxes):
+    """
+    https://adipandas.github.io/multi-object-tracker/_modules/motrackers/utils/misc.html
+    Calculate centroids for multiple bounding boxes.
+
+    Args:
+        bboxes (numpy.ndarray): Array of shape `(n, 4)` or of shape `(4,)` where
+            each row contains `(xmin, ymin, width, height)`.
+
+    Returns:
+        numpy.ndarray: Centroid (x, y) coordinates of shape `(n, 2)` or `(2,)`.
+
+    """
+
```

Comment:
I have a feeling this function is not very efficient.

Reply:
I borrowed it from another library. I choose it as it handles corner cases. Any suggestions?

Reply:
Maybe run on a long video and check how many time it costs. If it is all right, we can leave it as it is. We can think this as  the overhead of tracker UDF (not EVA).

Reply:
I see. We are benchmarking `eva` on larger datasets for correctness and performance, so will take care of it with that exercise.